lib/mix/tasks/compile/phoenix_live_view.ex

```diff
@@ -0,0 +1,110 @@
+defmodule Mix.Tasks.Compile.PhoenixLiveView do
+  @moduledoc """
+  A LiveView compiler for HEEx `:extract` cleanup.
+
+  You must add it to your `mix.exs` as:
+
+      compilers: Mix.compilers() ++ [:phoenix_live_view]
+
+  """
+  use Mix.Task
+
+  @recursive true
+  @manifest ".compile_live_view"
+  @manifest_version 2
+
+  @switches [
+    force: :boolean
+  ]
+
+  @doc false
+  def run(args) do
+    {compile_opts, _argv, _errors} = OptionParser.parse(args, switches: @switches)
+
+    {version, current_extracts} = read_manifest()
+    manifest_outdated? = version != @manifest_version or manifest_older?()
+
+    cond do
+      manifest_outdated? || compile_opts[:force] ->
+        remove_outdated_extracts(current_extracts || [])
+
+      true ->
+        {:noop, []}
+    end
+  end
+
+  defp remove_outdated_extracts(manifest_extracts) do
+    extracts = get_extracts(project_modules())
+
+    case manifest_extracts -- extracts do
+      [] ->
```

Comment:
there's one issue here in that if you include the extracted files in your git repo and then remove modules without ever compiling them, the files won't be removed